src/v/resource_mgmt/storage.cc

```diff
@@ -51,6 +51,9 @@ ss::future<> disk_space_manager::start() {
         _cache_disk_nid = _storage_node->local().register_disk_notification(
           node::disk_type::cache,
           [this](node::disk_space_info info) { _cache_disk_info = info; });
+        _data_disk_nid = _storage_node->local().register_disk_notification(
+          node::disk_type::data,
+          [this](node::disk_space_info info) { _data_disk_info = info; });
     }
```

Comment:
i think we are missing unregister part in `stop`

Reply:
thanks fixed. it was causing crash!